### PR TITLE
feat: export cjs interface for wdio-cli

### LINKED ISSUE #9869: [🐛 Bug]: CJS require of @wdio/cli and @wdio/reporter not implemented
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.4

### Node.js Version

16.13.2

### Mode

WDIO Testrunner

### Which capabilities are you using?

_No response_

### What happened?

Tests are written in a cjs project with the testrunner started programmatically. When upgrading to v8 some wdio modules are no longer possible to require. 
```
const Launcher = require('@wdio/cli').default
const WDIOReporter = require('@wdio/reporter').default;
```
fails since they only support esm import.
A temporary workaround is to use ```const { Launcher } = await import('@wdio/cli');```
but in the case of @wdio/reporter that is a bit trickier.

### What is your expected behavior?

To be able to use 
```
const Launcher = require('@wdio/cli').default
const WDIOReporter = require('@wdio/reporter').default;
```

### How to reproduce the bug.

Use 
```
require('@wdio/cli').default
```
or
```
require('@wdio/reporter').default
```
in v8

### Relevant log output

```typescript
Launcher failed to start the test Error [ERR_REQUIRE_ESM]: require() of ES Module
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### PR DESCRIPTION
**Pull request by DirkoOdendaal**
Fixes #9869 

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ x ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

Export interface for CJS

### CODE REVIEW

File: packages/wdio-cli/src/cjs/index.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
+import type { RunCommandArguments } from '../types.js'
+import type { Options, Capabilities } from '@wdio/types'
+import type { EndMessage } from '../launcher.js'
+
+export class Launcher {
+
+    esmLauncher: any
```
Comment (christian-bromann) [inline packages/wdio-cli/src/cjs/index.ts:7]:
```suggestion
    #esmLauncher: any
```

File: packages/wdio-cli/src/cjs/index.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    ) {
+        this.esmLauncher.startInstance(specs, caps, cid, rid, retries)
+    }
+
+    /**
+     * generates a runner id
+     * @param  {Number} cid capability id (unique identifier for a capability)
+     * @return {String}     runner id (combination of cid and test id e.g. 0a, 0b, 1a, 1b ...)
+     */
+    getRunnerId (cid: number) {
```
Comment (christian-bromann) [inline packages/wdio-cli/src/cjs/index.ts:85]: can we make a bunch of these methods private so we don't have to expose them here? The launcher has been build before private class functions in JavaScript were a thing. Maybe we can optimize the code here so we only have to expose the `run` method which is the only one we've documented.

Comment (DirkoOdendaal) [reply packages/wdio-cli/src/cjs/index.ts]: Some of the methods I refactored to private methods. Not sure how many people actually use this, so kept a few methods public for now.
Also happy making most of them private.

Comment (christian-bromann) [reply packages/wdio-cli/src/cjs/index.ts]: > Also happy making most of them private.

Let's make all of them private except `run`. We don't have anything else documented and therefor people should not use any other method. If folks start to report this, we can think about making them public again.

Comment (DirkoOdendaal) [reply packages/wdio-cli/src/cjs/index.ts]: Yeah you are right. Just did that.

**Review by christian-bromann — APPROVED** [review]
Nicely done 👌

**Pull request merged**